### PR TITLE
fix(ui): overriding user info after login

### PR DESCRIPTION
**Pull request by chirag-madlani**
### Describe your changes:

Fixes <issue-number>

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Review by ShaileshParmar11 — DISMISSED** [review]

File: openmetadata-ui/src/main/resources/ui/src/utils/UserDataUtils.ts [hunk 1]
```diff
@@ -124,3 +129,44 @@ export const getUserWithImage = (user: User) => {
...
+    updatedUserData.profile?.images &&
+    !matchUserDetails(existingUserDetails, updatedUserData, ['profile'])
+  ) {
+    const finalData = {
+      ...existingUserDetails,
+      //   We want to override any profile information that is coming from the OIDC provider
+      profile: {
+        ...existingUserDetails.profile,
+        profile: { ...existingUserDetails.profile, ...updatedUserData.profile },
+      },
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/UserDataUtils.ts:153]:
```suggestion
      profile: {
        ...existingUserDetails.profile,
        ...updatedUserData.profile
      },
```

**Review by Sachin-chaurasiya — DISMISSED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/Auth/AppAuthenticators/MsalAuthenticator.tsx [hunk 2]
```diff
@@ -104,12 +76,35 @@ const MsalAuthenticator = forwardRef<AuthenticatorRef, Props>(
...
+      handleOnLogoutSuccess();
+    };
+
     const fetchIdToken = async (
       shouldFallbackToPopup = false
     ): Promise<OidcUser> => {
       const tokenRequest = {
         account: account || accounts[0], // This is an example - Select account based on your app's requirements
-        scopes: msalLoginRequest.scopes,
+        scopes: ['User.Read'],
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Auth/AppAuthenticators/MsalAuthenticator.tsx:107]: @chirag-madlani , why are we changing this?

File: openmetadata-ui/src/main/resources/ui/src/utils/AuthProvider.util.ts [hunk 3]
```diff
@@ -223,7 +223,7 @@ export const getAuthConfig = (
 
 // Add here scopes for id token to be used at MS Identity Platform endpoints.
 export const msalLoginRequest: PopupRequest = {
-  scopes: ['openid', 'profile', 'email', 'offline_access'],
+  scopes: ['openid', 'profile', 'email', 'offline_access', 'User.Read'],
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/AuthProvider.util.ts:226]: @chirag-madlani, what is this for?

File: openmetadata-ui/src/main/resources/ui/src/components/Auth/AppAuthenticators/MsalAuthenticator.tsx (hunk 2 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/Auth/AppAuthenticators/MsalAuthenticator.tsx]: Was testing refreshToken flow and changed that but it didn't required so remove it. Thanks

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**